README.md
# poliklinikindah
 


about_chikitsa/CHANGELOG.md
# Change Log
All notable changes to this project will be documented in this file.
This project adheres to [Semantic Versioning](http://semver.org/).

## [2.0.7] - 2022-01-31
### Changed
-Bill form added fees tax.

## [2.0.6] - 2021-12-23
### Changed
-Minor Changes 

## [2.0.5] - 2021-12-03
### Changed
-Minor Changes 

## [2.0.4] - 2021-11-22
### Changed
-Minor Changes 

## [2.0.3] - 2021-11-18
### Changed
-Minor Changes 

## [2.0.2] - 2021-09-28
### Changed
-Minor Changes 

## [2.0.1] - 2021-09-14
### Changed
- Allow to Change Favicon.

## [2.0.0] - 2021-07-23
### Changed
Complete New Fresh Look.

## [1.0.4] - 2021-06-16
### Changed
-  Patient Account Enable/Disable

## [1.0.3] - 2021-04-27
### Changed
- Upload Language at time of Extension update and install
- Use Multicolumns Autocomplete

## [1.0.2] - 2021-04-21
### Changed
- Minor Corrections

## [1.0.1] - 2021-04-19
### Changed
- Faster Install
- Minor Changes

## [1.0.0] - 2021-04-03
### Changed
- Minor Corrections
## [0.9.9] - 2021-02-09
### Changed
- Medical History Report

## [0.9.8] - 2021-01-23
### Changed
- Minor Corrections

## [0.9.7] - 2021-01-19
### Changed
- Remove Auto Addition Of VAT Tax From Bill

## [0.9.6] - 2021-01-18
### Changed
- Added User Who Created Bill and User Who Collected Payments

## [0.9.5] - 2020-12-29
### Changed
- Add Vat Tax to Bill 
- Filter Bill Records

## [0.9.4] - 2020-11-11
### Changed
- Allow to edit Treatments in visit

## [0.9.3] - 2020-10-09
### Changed
- Minor Corrections

## [0.9.2] - 2020-06-16
### Added
- Show Added extensions in table
- Do not allow an extension to activate if license key is not valid
- If extension is already added, do not show on add extension page
- Extension must deactivate automatically if license key expires
- Allow to add Decimal Places, Decimal Symbol and Thousands Separator for Currency

### Changed
- Solved error in adding new language


## [0.9.1] - 2020-03-04
### Added
- Validation in Inquiry Form
- Patient History Report
- Separate Menu for Language Control
- Support for RTL Languages
### Changed
- Datatable in Payment Page
- Solved error in adding new language
- Solved problem in adding multiple days in Exceptional Days


## [0.9.0] - 2020-02-18
### Added
- User wise Prefered Language


## [0.8.9] - 2019-01-02
### Changed
- For Custom Detail 0.0.3
- Bill Bugs Solved

## [0.8.8] - 2019-12-18
### Added
- Use custom.css for customizing css
- Module Page faster

### Changed
- Treatment List problem solved

## [0.8.7] - 2019-12-13
### Changed
- Bill Form Responsive
- Upload only required languages to database


## [0.8.6] - 2019-12-13
### Changed
- Minor Corrections

## [0.8.5] - 2019-11-19
### Changed
- Language Changes

## [0.8.4] - 2019-10-02
### Changed
- Half Working Day
- Backup before update


## [0.8.3] - 2019-08-08
### Changed
- Minor Fixes
- Multiple Appointment overlap problem fixed

## [0.8.2] - 2019-08-06
### Changed
- Minor Fixes

## [0.8.1] - 2019-08-01
### Added
- Date Formats and Time Formats added
### Changed
- Minor Fixes
- Tax Calculation error solved
- Change Language File


## [0.8.0] - 2019-07-08
### Changed
- Minor Fixes

## [0.7.9] - 2019-07-02
### Added
  - Edit All Language Variables.
### Changed
  - Correction in Tax Calculations


## [0.7.8] - 2019-06-05
### Added
	- Config file will not replace on update
	- Language (English) file will not replace on update
	- Single Bill Form (instead of separate in Visit)
	- Display Change Log on Extension update


## [0.7.7] - 2019-05-19
### Added
- Allow to add Bill before Consultation
- Allow to add name of Patient in Regional Languages
- Added Aadhar Number / Unique ID to record Government Identifier for Patient.
- Instead of Email, Aadhar Number can be used to serach for Patient.
- Search Patient using Patient ID / Aadhar Number / Mobile Along with Name in Bill and Payment.
- Allow to add Doctor in Bill without Visit
- Show selected columns in Patient
- Add Inquiry on Inquiry Page
- Pending Payments - Some payment methods can be kept as pending .e.g. Cheque, Insurance.
  The Payment when done using this method will be kept as pending unless marked as Approved.
  If Payment is marked as Rejected, the amount will show as outstanding.
### Changed
- Problem in taking the last appointment of the day solved.
- If a visit is created for a appointment, clicking on the Appointment on Calendar will lead to Edit Visit.
- Add Patient will be in collapsed state while adding appointment
- Correction in Tax Calculation
- No hard codded Dr. Title for Doctor. Add the title while adding user.
- Minor Fixes

## [0.7.6] - 2019-03-19
### Added
- Browse Extensions from Chikitsa
- List Master
### Changed
- Minor Changes

## [0.7.5] - 2019-02-11
### Changed
- Minor Changes

## [0.7.4] - 2019-01-04
### Added
- Auto referesh of Appointment Page
- Issue Refund from Patient Account
### Changed
- Minor Changes

## [0.7.3] - 2018-10-29
### Added
- Payment Methods
### Changed
- Minor Fixes
- Patient Selection in Bill Report

## [0.7.2] - 2018-10-18
### Added
- Blood Group in Patient
- CSS and JS minification

### Changed
- Do not allow to delete a user if Doctor

## [0.7.1] - 2018-09-19
### Added
- Blood Group in Patient

### Changed
- Compatible with PHP 7.2.9
- Disabled calendar dates on Holidays
- Next Followup date will not set for Holiday
- Appointment cannot be taken on Holiday
- Better Patient Reference Details

## [0.7.0] - 2018-08-20
### Changed
- Better Language Settings
- Folder Structure improved

## [0.6.9] - 2018-08-10
### Changed
- made compatible with PHP 7.2.x
- Correction in Reference By
- Allowing to store data in different languages

## [0.6.8] - 2018-07-27
### Changed
- Correction in Tax Report
- Correction in Print Bill
- Other Minor fixes


## [0.6.7] - 2018-07-17
### Changed
- Correction in Tax Report


## [0.6.6] - 2018-07-09
### Added
- Filter in Bill
### Changed
- Taxes (Corrected problems)
- Tax Report
- Correction in Max Patient at a time logic
- Other minor fixes

## [0.6.5] - 2018-06-18

application/helpers/currency_helper.php
<?php
function get_number_format_settings(){
    //the database functions can not be called from within the helper
    //so we have to explicitly load the functions we need in to an object
    //that I will call ci. then we use that to access the regular stuff.
    $ci=& get_instance();
    $ci->load->database();
    
    

    //select the required fields from the database

    $ci->db->select('currency_symbol,number_of_decimal,decimal_symbol,thousands_separator,currency_postfix');

    
    
    //tell the db class the criteria
    $ci->db->where('invoice_id', 1);
    
    

    //supply the table name and get the data
    $query = $ci->db->get('invoice');
    
    

    foreach($query->result() as $row):
 
 

        //get the full name by concatinating the first and last names

        $number_format['currency_symbol'] = $row->currency_symbol;

        $number_format['number_of_decimal'] = $row->number_of_decimal;

        $number_format['decimal_symbol'] = $row->decimal_symbol;

        $number_format['thousands_separator'] = $row->thousands_separator;

        $number_format['currency_postfix'] = $row->currency_postfix;

 
 
    endforeach;
    
    

    return $number_format;

}







//convert numbers to currency format
if ( ! function_exists('currency_format'))
{
  function currency_format($number)
  {

	/*$number_format = get_number_format_settings();

    $currencySymbol = $number_format['currency_symbol'];

    $decimalPlaces = $number_format['number_of_decimal'];

    $decimalSymbol = $number_format['decimal_symbol'];

    $thousandsSeparator = $number_format['thousands_separator'];

	$currencyPostfix = $number_format['currency_postfix'];
	
	*/
	$ci = get_instance(); // CI_Loader instance
	
	$currencySymbol = $ci->config->item('currencySymbol');
	$decimalPlaces =  $ci->config->item('decimalPlaces');
	$decimalSymbol = $ci->config->item('decimalSymbol');
	$thousandsSeparator = $ci->config->item('thousandsSeparator');
	$currencyPostfix =  $ci->config->item('currencyPostfix');

		return $currencySymbol. number_format($number, $decimalPlaces, $decimalSymbol, $thousandsSeparator).$currencyPostfix;

  }
}

function numberTowords($num){ 
	$ones = array( 
		0 => "",
		1 => "One", 
		2 => "Two", 
		3 => "Three", 
		4 => "Four", 
		5 => "Five", 
		6 => "Six", 
		7 => "Seven", 
		8 => "Eight", 
		9 => "Nine", 
		10 => "Ten", 
		11 => "Eleven", 
		12 => "Twelve", 
		13 => "Thirteen", 
		14 => "Fourteen", 
		15 => "Fifteen", 
		16 => "Sixteen", 
		17 => "Seventeen", 
		18 => "Eighteen", 
		19 => "Nineteen" 
	); 
	$tens = array( 
		0 => "",
		1 => "Ten",
		2 => "Twenty", 
		3 => "Thirty", 
		4 => "Forty", 
		5 => "Fifty", 
		6 => "Sixty", 
		7 => "Seventy", 
		8 => "Eighty", 
		9 => "Ninety" 
		); 
		
		

	$hundreds = array( 
		"Hundred", 
		"Thousand", 
		"Million", 
		"Billion", 
		"Trillion", 
		"Quadrillion" 
		); 
	$num = number_format($num,2,".",","); 
	$num_arr = explode(".",$num); 
	$wholenum = $num_arr[0]; 
	$decnum = $num_arr[1]; 
	$whole_arr = array_reverse(explode(",",$wholenum)); 
	krsort($whole_arr); 
	$rettxt = ""; 
	foreach($whole_arr as $key => $i){ 
		if($i < 20){ 
			$rettxt .= $ones[$i]; 
		}elseif($i < 100){ 
			$rettxt .= $tens[substr($i,0,1)]; 
			$rettxt .= " ".$ones[substr($i,1,1)]; 
		}else{ 
			$rettxt .= $ones[substr($i,0,1)]." ".$hundreds[0]; 
			$rettxt .= " ".$tens[substr($i,1,1)]; 
			$rettxt .= " ".$ones[substr($i,2,1)]; 
		} 
		if($key > 0){ 
			$rettxt .= " ".$hundreds[$key]." "; 
		} 
	} 
	if($decnum > 0){ 
		$rettxt .= " and "; 
		if($decnum < 20){ 
			$rettxt .= $ones[$decnum]; 
		}elseif($decnum < 100){ 
			$rettxt .= $tens[substr($decnum,0,1)]; 
			$rettxt .= " ".$ones[substr($decnum,1,1)]; 
		} 
	} 
	return $rettxt; 

}
?>